ruff_lsp/server.py

```diff
@@ -736,47 +1023,116 @@ async def format_document(
         ls: server.LanguageServer,
         params: DocumentFormattingParams,
     ) -> list[TextEdit] | None:
-        uri = params.text_document.uri
-        document = ls.workspace.get_text_document(uri)
-        return await _format_document_impl(document)
-
-
-async def _format_document_impl(
-    document: workspace.TextDocument,
-) -> list[TextEdit]:
-    result = await _run_format_on_document(document)
-    return _result_to_edits(document, result)
+        # For a Jupyter Notebook, this request can only format a single cell as the
+        # request itself can only act on a text document. A cell in a Notebook is
+        # represented as a text document.
+        document = Document.from_text_document(
+            ls.workspace.get_text_document(params.text_document.uri)
+        )
+        result = await _run_format_on_document(document)
+        if result is None:
+            return None
+        return _fixed_source_to_edits(
+            original_source=document.source,
+            fixed_source=result.stdout.decode("utf-8"),
+            is_notebook_file=document.is_notebook_file(),
+        )
 
 
 async def _fix_document_impl(
-    document: workspace.TextDocument,
+    document: Document,
     *,
     only: str | None = None,
-) -> list[TextEdit]:
-    result = await _run_check_on_document(document, extra_args=["--fix"], only=only)
-    return _result_to_edits(document, result)
+) -> WorkspaceEdit | None:
+    result = await _run_check_on_document(
+        document,
+        extra_args=["--fix"],
+        only=only,
+    )
+    return _result_to_workspace_edit(document, result)
 
 
-def _result_to_edits(
-    document: workspace.TextDocument,
-    result: RunResult | None,
-) -> list[TextEdit]:
+def _result_to_workspace_edit(
+    document: Document, result: RunResult | None
+) -> WorkspaceEdit | None:
+    """Converts a run result to a WorkspaceEdit."""
```

Comment:
Source level code actions should apply the fix for _every_ cell in a Notebook. Earlier the API was to create a single text edit to replace the entire content of a text document with the fixed source, create a workspace edit corresponding the text edit and send it back. This behavior is preserved when the document kind is Python.

But, for a Notebook, we need to create text edits for each cell where the fix exists which is why this was changed.